app/models/current.rb

```diff
@@ -6,7 +6,7 @@ class Current < ActiveSupport::CurrentAttributes
   def self.initialize_with(client: nil)
     self.client = client
 
-    if client&.authenticated?
+    if client&.is_a?(Client) && client&.authenticated?
```

Comment:
This could be simplified to `if client.is_a?(Client) && client.authenticated?`. `nil.is_a?(AnythingButNilClass)` will be `false` (assuming no meta-programming).

Reply:
Interesting, that .nil syntax is new to me. I’m intrigued.

But I don’t quite follow your train of thought. What’s the thing you’re simplifying?

Reply:
Good catch, I just removed the &. For awhile I thought this nil check was all I needed but then I ran into one weird situation where something other than a client was passed into here. To be honest, maybe I should have caught that in an alternate way but my fix was to confirm that client is indeed a client. But with that in place, the nil check was no longer needed.

thanks for pointing out

Reply:
Actually, I take that back. I should really fix this properly. The fact that a non-Client was being passed in is a deeper code small. I'm fixing the type mismatch at the root so then I don't need this is_a?(Client) check